torch/testing/_internal/common_utils.py

```diff
@@ -2673,6 +2673,55 @@ def assertAtenOp(self, onnx_model, operator, overload_name=""):
         self.assertEqual(attrs["operator"], operator)
         self.assertEqual(attrs.get("overload_name", ""), overload_name)
 
+    def check_nondeterministic_alert(self, fn, caller_name, should_alert=True):
+        '''Checks that an operation produces a nondeterministic alert when
+        expected while `torch.use_deterministic_algorithms(True)` is set.
+
+        Args:
+          fn (callable): Function to check for a nondeterministic alert
+
+          caller_name (str): Name of the operation that produces the
+              nondeterministic alert. This name is expected to appear at the
+              beginning of the error/warning message.
+
+          should_alert (bool, optional): If True, then the check will only pass
+              if calling `fn` produces a nondeterministic error/warning with the
+              expected message. If False, then the check will only pass if
+              calling `fn` does not produce an error. Default: `True`.
+        '''
+
+        error_message = '^' + caller_name + ' does not have a deterministic implementation, but you set'
+
+        # Check that errors are thrown correctly
+        with DeterministicGuard(True):
+            if should_alert:
+                with self.assertRaisesRegex(
+                        RuntimeError,
+                        error_message,
+                        msg='expected a non-deterministic error, but it was not raised'):
+                    fn()
+
+            else:
```

Comment:
Interesting lack of parallelism between this if/else block and the following for warnings. Why handle the error not thrown case here but not the warning not thrown case?

Reply:
Good point, I'll add the not thrown case

Reply:
I've added it